src/components/i-calendar.vue

```diff
@@ -0,0 +1,537 @@
+        }
+      },
+    },
+    value: {
+      handler(val) {
+        const valueArray = Array.isArray(val) ? val : [val];
+
+        const isValueSame = this.checkSame(valueArray, this.selectedDate);
+        if (!isValueSame) {
+          this.selectedDate = valueArray.filter((date) => !!date).map((date) => dayjs(date.toString()).toDate());
+        }
+      },
+    },
+  },
+  created() {
+    this.activeDate = this.initialDate;
+  },
+  mounted() {
+    if (this.value) {
+      if (Array.isArray(this.value)) {
+        if (this.value.length > 0) {
+          const [first] = this.value;
+          this.activeDate = dayjs(first.toString()).toDate();
+          const results = this.value.map((date) => dayjs(date.toString()).toDate());
+          this.selectedDate = results;
+        } else {
+          this.activeDate = dayjs().toDate();
+        }
+      } else {
+        this.activeDate = dayjs(this.value ? this.value.toString() : null).toDate();
+        this.selectedDate.push(this.activeDate);
+      }
+    } else if (!this.activeDate) {
+      this.activeDate = dayjs().toDate();
+      this.clickDate(this.activeDate);
+    }
+  },
+  methods: {
+    checkSame(array1, array2) {
+      const isSame =
+        array1.length === array2.length &&
+        array1.every((element, index) => dayjs(element).isSame(array2[index], 'day'));
+      return isSame;
+    },
+    clickDate(date) {
+      const findMatchIndex = this.selectedDate.findIndex((d) => dayjs(date).isSame(dayjs(d), 'day'));
+      if (findMatchIndex === -1) {
+        this.selectedDate.shift();
+        this.selectedDate.push(dayjs(date).second(0).toDate());
+        this.$emit('onSelectDate');
+      } else {
+        this.selectedDate.splice(findMatchIndex, 1);
+      }
+    },
+    clickMonth(month) {
+      this.activeDate = dayjs(month.monthValue.toString()).toDate();
+      this.showMonthYearTable = false;
+    },
+    isSelectedDate(date) {
+      return this.selectedDate.find((d) => dayjs(date).isSame(dayjs(d), 'day'));
+    },
+    isSelectedMonth(date) {
+      return dayjs(date).isSame(dayjs(this.activeMonthPicker), 'month');
+    },
+    clickPreviousMonth() {
+      this.activeDate = dayjs(this.activeDate.toString()).subtract(1, 'month').toDate();
+    },
+    clickNextMonth() {
+      this.activeDate = dayjs(this.activeDate.toString()).add(1, 'month').toDate();
+    },
+    clickPreviousYear() {
+      this.activeYearPicker = dayjs(this.activeYearPicker.toString()).subtract(1, 'year').toDate();
+    },
+    clickNextYear() {
+      this.activeYearPicker = dayjs(this.activeYearPicker.toString()).add(1, 'year').toDate();
+    },
+    clickMonthPicker() {
+      this.activeMonthPicker = this.activeDate;
+      this.activeYearPicker = dayjs(this.activeDate.toString()).format('YYYY');
+      this.showMonthYearTable = !this.showMonthYearTable;
+    },
+  },
+};
+</script>
+
+<style>
+.i-calendar {
+  width: 735px;
```

Comment:
consider to set the width to 100%.. let the parent side control the width

Reply:
ok